substrate/primitives/metadata-ir/src/types.rs

```diff
@@ -138,6 +138,8 @@ pub struct PalletMetadataIR<T: Form = MetaForm> {
 	pub index: u8,
 	/// Pallet documentation.
 	pub docs: Vec<T::String>,
+	/// Deprecation info
+	pub deprecation_info: DeprecationStatus<T>,
```

Comment:
Thinking about the other PRs, what's the actual value this can hold for pallet metadata? It would only be possible to hold `FullyDeprecated` or `NotDeprecated`?

Reply:
NotDeprecated,deprecatedWithoutNote and Deprecated(note,since). I'm also a bit on the fence whether it makes sense to deprecate the whole pallet or not